pkg/logging/recorded.go

```diff
@@ -64,3 +67,49 @@ func WithRecorder(logger kitlog.Logger, recorder record.EventRecorder, object ru
 		},
 	)
 }
+
+// RecorderAwareCaller returns the file and line where the Log method was
+// invoked, adjusting for the fact that this may have been hijacked by the
+// event recorder.
+func RecorderAwareCaller() kitlog.Valuer {
```

Comment:
I felt like testing this would be a bit of a nightmare, so didn't. Other opinions welcome!

Reply:
Effort vs impact- we'll know if this breaks and it only inconveniences people. Happy without tests.